File: packages/formspec-layout/src/defaults.ts
/**
 * Map a definition item to its default component type based on `dataType`.
 *
 * Used as a fallback when no component document is provided or when the
 * theme's widget cascade doesn't resolve to an available component.
 *
 * @param item - A definition item with a `dataType` property.
 * @returns The default component type string (e.g. "TextInput", "NumberInput").
 */
export function getDefaultComponent(item: { dataType?: string }): string {
    switch (item.dataType) {
        case 'string': return 'TextInput';
        case 'text': return 'TextInput';
        case 'integer':
        case 'decimal':
        case 'number': return 'NumberInput';
        case 'boolean': return 'Toggle';
        case 'date': return 'DatePicker';
        case 'dateTime': return 'DatePicker';
        case 'time': return 'DatePicker';
        case 'uri': return 'TextInput';
        case 'choice': return 'Select';
        case 'multiChoice': return 'CheckboxGroup';
        case 'attachment': return 'FileUpload';
        case 'money': return 'NumberInput';
        default: return 'TextInput';
    }
}

File: crates/formspec-plan/src/defaults.rs
//! Default component mapping from dataType to component type.

/// Returns the default component type for a given Formspec dataType string.
pub fn get_default_component(data_type: &str) -> &'static str {
    match data_type {
        "string" | "text" => "TextInput",
        "integer" | "decimal" | "number" => "NumberInput",
        "boolean" => "Toggle",
        "date" | "dateTime" | "time" => "DatePicker",
        "uri" => "TextInput",
        "choice" => "Select",
        "multiChoice" => "CheckboxGroup",
        "attachment" => "FileUpload",
        "money" => "MoneyInput",
        _ => "TextInput",
    }
}
